### DISCUSSION TITLE
Is there a way to access the negative binomial distribution parameters from the sceptre object

### DISCUSSION THREAD

**Discussion by jamesgalante** [Katsevich-Lab/sceptre#118] [Q&A] [answered]
I want to see the negative binomial distribution fit to each response (gene). Specifically the NB distr fit to the null control cells for a specific gene. Does Sceptre calculate this explicitly and can it be accessed?

**Comment by timothy-barry** [comment]
Hi James,

Yes, you can access this. Are you working with high-MOI data? If so, `sceptre_object@response_precomputations` stores the per-gene fitted coefficients and theta from an NB regression of gene expression onto the *entire* set of cells (tha passed QC). Also, you can obtain the covariate matrix used in this regression via `sceptre_object@covariate_matrix`. Finally, you can obtain the response matrix via `get_response_matrix(sceptre_object)`. These ingredients should enable you to redo the regression on your own.

**Reply by jamesgalante** [reply]
Thanks! If I'm working with low MOI data does this change?

**Comment by timothy-barry** [comment]
Yes, it changes a bit depending on what the [control group](https://timothy-barry.github.io/sceptre-book/set-analysis-parameters.html#sec-set-analysis-parameters_control_group) is. If `control_group` is set to `"complement"`, then the interpretation is exactly the same as the above. However, if `control_group` is set to `"nt_cells"`, then the per-gene regression coefficients in `sceptre_object@response_precomputations` correspond to an NB regression of gene expression onto the set of cells harboring an NT gRNA only.

**Reply by jamesgalante** [reply]
Awesome thank you! -> is there a reason you don't compute the NB distr on the entire set of complement cells as opposed to the entire set of cells that passed qc?

**Reply by timothy-barry** [reply]
This is a bit subtle, but in high-MOI, we fit a single NB regression model for each gene. This NB regression model is then shared across all gRNAs tested for association against that gene. This is just the standard regression paradigm. If I want to test for association between a binary "treatment" vector $(X_1, \dots, X_n)$ and a response vector $(Y_1, \dots, Y_n)$, then I would regress the entire response vector onto the entire treatment vector. It might be most efficient to discuss verbally if you'd be interested in more info.

**Comment by timothy-barry** [comment, marked as answer]
Closing. Feel free to drop additional questions if you have them.
